src/Native/CpuMathNative/Sse.cpp

```diff
@@ -532,7 +533,8 @@ EXPORT_API(float) Sum(const float* pValues, int length)
             result = _mm_add_ps(result, temp);
 
             pValues += misalignment;
-            length -= misalignment;
+			// safe to downcastcast as misalignment < 16.
+			length -= static_cast<int>(misalignment);
```

Comment:
>static_cast [](start = 13, length = 11)

same here, you can use (int)misalignment;

Reply:
I thought in C++ we are to use *_cast<>() in most cases as per guidelines... ([http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts))

---
In reply to: [276761019](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276761019) [](ancestors = 276761019)

Reply:
ok thanks Gleb, I wont argue with Herb Sutter

---
In reply to: [276795215](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276795215) [](ancestors = 276795215,276761019)